### PR TITLE
remote_file support for windows network shares

### PR DESCRIPTION
**Pull request by jdmundrawala**
```ruby
remote_file 'C:\Foo.tar.gz' do
  source "\\\\foohost\\fooshare\\Foo.tar.gz"
end
```

cc @btm @ksubrama @smurawski

### CODE REVIEW

File: lib/chef/provider/remote_file/fetcher.rb [hunk 1]
```diff
@@ -23,15 +23,29 @@ class RemoteFile
...
+              raise ArgumentError, "Invalid uri, Only http(s), ftp, and file are currently supported"
+            end
+          end
+        end
+
+        # Windows network share: \\computername\share\file
+        def self.network_share?(source)
+          case source
+          when String
+            !!(%r{\A\\\\[A-Za-z][A-Za-z0-9+\-\.]*} =~ source)
```
Comment (ksubrama) [inline lib/chef/provider/remote_file/fetcher.rb]: I believe \\192.168.x.x is a valid way to reach a share (you might want to double check that).  In that case, we should allow numbers in the first character.

Comment (jdmundrawala) [inline lib/chef/provider/remote_file/fetcher.rb]: Do we have to support that? Is it different than '\\192.168.x.x'?
Also, is `\fooshare` valid?

Comment (ksubrama) [inline lib/chef/provider/remote_file/fetcher.rb]: Sorry - I meant \\IPaddress

https://msdn.microsoft.com/en-us/library/gg465305.aspx

Comment (jdmundrawala) [inline lib/chef/provider/remote_file/fetcher.rb]: oh i see

**Comment by ksubrama** [comment]
:+1:

**Comment by smurawski** [comment]
:+1:

**Pull request merged**